src/gui/src/SongPropertiesDialog.h

```diff
@@ -32,7 +32,7 @@
  */
 class SongPropertiesDialog : public QDialog, private Ui_SongPropertiesDialog_UI
 {
-    H2_OBJECT
+    H2_OBJECT(SongPropertiesDialog)
```

Comment:
This one is odd. The class itself does not inherit from `H2Core::Object`. If think it's save to remove the macro.

Reply:
done

Reply:
I wonder whether I have been a little premature on this one. Can you think of a reason why any of the classes defined in src/core and src/gui should _not_ be based on the `Object` template? In terms of debugging, especially detecting memory leakage, it would certainly be useful to derive them all.

Reply:
Yes, you are right, now that H2Core::Object is really lightweight, all Hydrogen classes should inherit from it, so we can actually track instances accurately. I think we could merge this PR in development branch, and complete the work of making all classes from gui inherit H2Core::Object as we touch/refactor them. What do you think ?

Reply:
> I think we could merge this PR in development branch, and complete the work of making all classes from gui inherit H2Core::Object as we touch/refactor them. What do you think ?

I would prefer doing this for all classes at the same time. It would take quite a while to refactor/touch all the classes and there are already so many other things to do (we have a ratio of open to solved issues of 1 : 3. There is still a lot of room at the top.)